### ISSUE TITLE
[qt6-static] QSerialPort not found

### ISSUE THREAD

**Issue by Sonic-Amiga** [msys2/MINGW-packages#19890]
### Description / Steps to reproduce the issue

I have a project made in QTCreator and i want a static Windows build, so that i could give .exe to other people with no problems. But after choosing a kit named "Desktop Qt 6.6.0 (static) MinGW-w64 MINGW64 (MSYS2)" i can't complete CMake configure step:
```
-- Could NOT find Qt6SerialPort (missing: Qt6SerialPort_DIR)
CMake Error at CMakeLists.txt:13 (find_package):
  Found package configuration file:

    C:/msys64/mingw64/qt6-static/lib/cmake/Qt6/Qt6Config.cmake

  but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
  FOUND.  Reason given by package:

  Failed to find required Qt component "SerialPort".

  Expected Config file at
  "C:/msys64/mingw64/qt6-static/lib/cmake/Qt6SerialPort/Qt6SerialPortConfig.cmake"
  does NOT exist
```
 I looked into /mingw64/qt6-static/lib, indeed, there is no libQT6SerialPort*.*

Just for your reference, project is here: https://github.com/Sonic-Amiga/ESP8266-Faikin/tree/main/Simulators/CN_WIRED/ACSimulator . I am a QT n00b, so i didn't write any single CMakeLists line by hand.

### Expected behavior

The app compiles normally

### Actual behavior

CMake error (see description)

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

Sorry, i am no expert in building these huge things

**Comment by MehdiChinoune** [comment]
qt6-static takes a long time to be built, so we decided to build only some submodules https://github.com/msys2/MINGW-packages/blob/0b6aefea8/mingw-w64-qt6-static/PKGBUILD#L191

**Comment by MehdiChinoune** [comment]
```
I have a project made in QTCreator and i want a static Windows build, so that i could give .exe to other people with no problems.
```
What problems will people have if you dynamically link your app and distribute it with the required dlls?
Many Programs alrady do that.

**Comment by Sonic-Amiga** [comment]
Well, first of all, i have problems. Is there a simple way to create some sort of a "redistributable bundle", i. e. some folder with the .exe and its accompanying DLLs ? I am not aware of such a thing.

**Comment by MehdiChinoune** [comment]
> Well, first of all, i have problems. Is there a simple way to create some sort of a "redistributable bundle", i. e. some folder with the .exe and its accompanying DLLs ? I am not aware of such a thing.

1- I am talking about problems for people you are distributing your app.
2- Have you ever heard of windeployqt!
3- Check your app runtime dlls and copy them to the same folder as your app.

Search Google.

**Comment by Sonic-Amiga** [comment]
No, i haven't heard of windeployqt. As i said, i am a total n00b in qt. sorry for that. I thought that static linkage is an ultimate solution.

**Comment by Sonic-Amiga** [comment]
Tried windeployqt, thank you for hints; yet it doesn't do all it should.
It perfectly detects QT DLLs and copies them over; yer these DLLs depend on:
libgcc_s_seh-1.dll
libstdc++-6.dll
libwinpthread-1.dll
--compiler-runtime option, according to the documentation, should do this, but it doesn't help.

> Check your app runtime dlls and copy them to the same folder as your app.

Is there any tool for that ? I'd expect it to be readily available in MinGW, the task seems pretty common.
